### ISSUE TITLE
Support constraints for computed properties

### ISSUE THREAD

**Issue by CodesInChaos** [edgedb/edgedb#1622]
Currently SDL like this does not work:

```
type Test {
	required property name -> str;
	property lower_name := str_lower(.name) {
		constraint exclusive;
	}
}
```

The functionality appears to be equivalent to #1164, but for consistency with normal properties it'd be nice if specifying them on computed properties instead of via `constraint on` on the object was possible.

**Comment by msullivan** [comment]
This is supported via `using`. The core bits were around before, but I fixed up problems with it in #2188.

**Comment by elprans** [comment]
Example:

```
type Test {
	required property name -> str;
	property lower_name {
            using (str_lower(.name));
            constraint exclusive;
	}
}
```
